dev/sbt-plugin/src/main/scala/com/lightbend/lagom/sbt/LagomPlugin.scala

```diff
@@ -486,17 +487,24 @@ private[sbt] object ConsoleHelper {
           override def close(): Unit = services.foreach(_.close())
         })
       case _ =>
+        import scala.concurrent.ExecutionContext.Implicits.global
+        import scala.concurrent.duration._
         // blocks until user press CTRL+D
         interaction.waitForCancel()
         // then shut down all running services
         log.info("Stopping services")
-        services.foreach { service =>
-          print(".")
-          service.close()
-        }
+
+        //Stop services in asynchrone manner
+        val closing = Future.traverse(services)(serv => Future {
+          serv.close()
+        })
+        closing.onComplete(_ => println("All services are stopped"))
+        Await.result(closing, 60 seconds)
+
```

Comment:
Hi @jroper 

Many thanks for the details in your answer. they help me to understand the problems with my solution.

If I understood your explanations, there is none simple solution to fix this issue ?
When you are talking about a dedicated thread pool, you mean about specific execution context? It is conceivable to take this option according to you?

Yes it should be interesting to try to modify the Play's ServerWithStop even if this require a Play new major release (and very long time to process it).

Many thanks,

Fabrice